### PR TITLE
Gateway nonce now starts at zero

### PR DESCRIPTION
**Pull request by deepesh-kn**
Gateway nonce now starts at zero, 
- Updated code
- Updated test proof data 
- Updated unit tests

Fixes: #700

### CODE REVIEW

**Comment by schemar** [comment]
Reviewing 👁

File: test/gateway/eip20_cogateway/redeem.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ const MockToken = artifacts.require('MockToken');
 
 const { MessageStatusEnum } = messageBus;
 
-const nonce = new BN(1);
+const nonce = new BN(0);
```
Comment (schemar) [inline test/gateway/eip20_cogateway/redeem.js]: why hardcode it and not get it from the gateway?

File: test/gateway/eip20_cogateway/confirm_stake_intent.js [hunk 2]
```diff
@@ -275,25 +275,6 @@ contract('EIP20CoGateway.confirmStakeIntent() ', (accounts) => {
     );
   });
 
-  it('should fail when staker nonce is already consumed.', async () => {
```
Comment (schemar) [inline test/gateway/eip20_cogateway/confirm_stake_intent.js]: Why did you delete this test? Shouldn't it still be tested? (e.g. try to use nonce `0` twice in a row)

File: test/gateway/eip20_gateway/confirm_redeem_intent.js [hunk 3]
```diff
@@ -326,24 +326,6 @@ contract('EIP20Gateway.confirmRedeemIntent()', (accounts) => {
     );
   });
 
-  it('should fail when redeemer nonce is already consumed', async () => {
```
Comment (schemar) [inline test/gateway/eip20_gateway/confirm_redeem_intent.js]: Why did you delete this test? Shouldn't it still be tested? (e.g. try to use nonce 0 twice in a row)

File: test/gateway/eip20_gateway/stake.js [hunk 4]
```diff
@@ -234,7 +234,7 @@ contract('EIP20Gateway.stake() ', (accounts) => {
     await mockToken.approve(gateway.address, stakeAmount, { from: stakerAddress });
     await baseToken.approve(gateway.address, bountyAmount, { from: stakerAddress });
 
-    nonce = new BN(2);
+    nonce = new BN(1);
```
Comment (schemar) [inline test/gateway/eip20_gateway/stake.js]: hardcoded on purpose? Why not "get nonce"?

File: test/gateway/eip20_gateway/stake.js [hunk 5]
```diff
@@ -259,7 +259,7 @@ contract('EIP20Gateway.stake() ', (accounts) => {
...
```
Comment (schemar) [inline test/gateway/eip20_gateway/stake.js]: hardcoded on purpose? Why not "get nonce"?

File: test/gateway/eip20_gateway/stake.js (hunk 4 above)
Comment (deepesh-kn) [reply test/gateway/eip20_gateway/stake.js]: It is an existing code. 
I am in favor of using hardcoded values, as we know what should be the nonce.
Any change in the contract logic will break the test.

Why do you think that it should get from "get nonce" ?

File: test/gateway/eip20_gateway/confirm_redeem_intent.js (hunk 3 above)
Comment (deepesh-kn) [reply test/gateway/eip20_gateway/confirm_redeem_intent.js]: Similarly,
This test is already covered in `should fail to confirm redeem intent if its already confirmed once`.
In fact, this test was invalid as the redeemer nonce always started with 1.

File: test/gateway/eip20_cogateway/confirm_stake_intent.js (hunk 2 above)
Comment (deepesh-kn) [reply test/gateway/eip20_cogateway/confirm_stake_intent.js]: This test is already covered in `should fail to confirm stake intent if its already confirmed once`.
In fact, this test was invalid as the staker nonce always started with 1.

File: test/gateway/eip20_cogateway/redeem.js (hunk 1 above)
Comment (deepesh-kn) [reply test/gateway/eip20_cogateway/redeem.js]: Why not hardcoded? 

We already know that the nonce will start at 0. 
This will also assert that nonce starts at 0.

Comment (schemar) [reply test/gateway/eip20_cogateway/redeem.js]: ok

File: test/gateway/eip20_cogateway/confirm_stake_intent.js (hunk 2 above)
Comment (schemar) [reply test/gateway/eip20_cogateway/confirm_stake_intent.js]: ok
